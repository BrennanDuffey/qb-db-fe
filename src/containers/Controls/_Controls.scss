.controls-section {
  width: 40%;
  font-size: 1.25rem;
  .filter-controls {
    display: flex;
    flex-direction: column;
    margin-top: 25px;
    p {
      transform: translate(-10%);
      margin-bottom: 0;
    }
  }
  .count-controls {
    display: flex;
    flex-direction: column;
    width: 30.3%;
    margin: 34px auto;
  }
  .buttons {
    display: flex;
    justify-content: space-around;
    margin-top: 100px;
    button {
      padding: 10px 20px;
      background-color: rgba(0, 0, 0, .1); 
      border-radius: 5px;
      font-size: 1.5rem;
    }
  }
}