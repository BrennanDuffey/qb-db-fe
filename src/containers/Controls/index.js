import React, { Component } from 'react';
import Select from 'react-select';
import 'bootstrap/dist/css/bootstrap.min.css';

class Controls extends Component {
  constructor() {
    super();
    this.state = {
      selectedCategories: []
    };
  };

  selectCategories = (selectedOptions) => {
    const selectedCategories = selectedOptions.map(option => `${option.value}`)
    this.setState({ selectedCategories })
  }

  render() {
    const categories = [
      { label: 'Geography', value: 20 },
      { label: 'History', value: 18 },
      { label: 'Science', value: 17 },
      { label: 'Literature', value: 15 },
      { label: 'Fine Arts', value: 21 },
      { label: 'Religion', value: 19 },
      { label: 'Mythology', value: 14 },
      { label: 'Philosophy', value: 25 },
      { label: 'Current Events', value: 26 },
      { label: 'Social Science', value: 22 }
    ];

    return (
      <section>
        <div className="container">
          <div className="row">
            <div className="col-md-4"></div>
            <div className="col-md-4">
              <Select options={ categories } isMulti onChange={(e) => {this.selectCategories(e)}} />
            </div>
            <div className="col-md-4"></div>
          </div>
        </div>
      </section>
    )
  };
};

export default Controls;