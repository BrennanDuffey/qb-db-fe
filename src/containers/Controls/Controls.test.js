import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { Controls } from '../Controls';

describe('Controls', () => {
  describe('Controls component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Controls />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});