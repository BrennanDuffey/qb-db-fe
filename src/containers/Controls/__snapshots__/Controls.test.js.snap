// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Controls component should match snapshot 1`] = `ShallowWrapper {}`;
